pkg/services/comments/commentmodel/permissions.go

```diff
@@ -104,6 +105,12 @@ func (c *PermissionChecker) CheckWritePermissions(ctx context.Context, orgId int
 		if !c.features.IsEnabled(featuremgmt.FlagAnnotationComments) {
 			return false, nil
 		}
+		if c.features.IsEnabled(featuremgmt.FlagAccesscontrol) {
```

Comment:
A couple of questions due to my poor understanding of permissions 🙏 

Previously we looked at whether user can edit dashboard to write a comment for annotation. If `featuremgmt.FlagAccesscontrol` is enabled - should we still check dashboard edit permissions based on guardian below? I mean that if access control returns `true` here then we still checking dashboard edit permission below. Is it OK?

Also, should we add similar checks for `PermissionChecker.CheckReadPermissions` above? (check read permissions determine whether user can load annotation related comments, this happens at the moment of showing an annotation on panel)

Reply:
Great questions! We're also still figuring FGAC out, so were asking ourselves the same things 😄 

Regarding the first question - yes, we should still keep the dashboard permission check. We were going back and forth on it, but eventually decided that for now the user needs both - the permission to edit annotations and the permission to edit the specific dashboard - to edit an annotation. If there are customer use cases where they want users to be able to edit annotations but not dashboards, this might change.

Regarding the second question - we've added a filter to annotation listing endpoint, so only annotations that the user is allowed to read are returned by annotation `Find` method. So you're already running the FGAC check by calling `Find`. As Gabriel mentioned in his comment, we might extend `Find` method to also allow filtering for annotations that you have write access to, in which case we will also be able to remove the check that I'm adding in this PR.

Hope that makes sense!

Reply:
`accesscontrol.ActionAnnotationsRead` will be checked from annotation `Find` method - a filter has been added in this [PR](https://github.com/grafana/grafana/pull/47467/files#diff-5a382284915b31e0cfcc606446bcd3e38ad97c26fce35af6081bbdbc8c8c8d0aR259). So you'll receive a `missing permissions` error from `CheckReadPermissions` which calls [`Find`](https://github.com/grafana/grafana/blob/73a7e8c7700ce70dcda713c35f0317a27071a5d3/pkg/services/comments/commentmodel/permissions.go#L66), if you try to list annotations but don't have access to read them.